### PR TITLE
Added configuration yml file for github actions

### PR DESCRIPTION
**Pull request by krhitesh**
## Fixes

Related #245  by @dhruvkb 

## Description

Picking up from the description in the issue:
> GitHub Actions offers a free and unlimited CI package for open source repositories with features rivalling and integrations surpassing those offered by CircleCI. The goal is to move our CI pipeline from CircleCI to GitHub Actions to run tests on all branches and pull requests.

This PR contains the github actions workflow configuration yml file. The workflow executes three jobs post every push or pull request. These jobs are **lint**, **build** and **test** similar to those in CircleCI configuration.

## Technical details
NPM dependencies are cached on pre-execution of **lint** job in ~/.npm directory and retrieved on execution of subsequent jobs. [Source](https://help.github.com/en/actions/language-and-framework-guides/using-nodejs-with-github-actions#example-caching-dependencies).

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .github/workflow/config.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: build
+      uses: actions/setup-node@v1
+      with:
+        node-version: '8.12.0'
+    - run: npm install
+    - run: npm run test
+      env:
+        CI: true
```
Comment (dhruvkb) [inline .github/workflow/config.yaml]: Isn't this the same as `test`?

File: .github/workflow/config.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  test:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: test
+      uses: actions/setup-node@v1
+      with:
+        node-version: '8.12.0'
```
Comment (dhruvkb) [inline .github/workflow/config.yaml]: Could do with Node v12. v8 is way too old.

File: .github/workflow/config.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        CI: true
+
+  test:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: test
+      uses: actions/setup-node@v1
```
Comment (dhruvkb) [inline .github/workflow/config.yaml]: @megha070 a good takeaway would be the use of predefined actions. See [GitHub Actions](https://github.com/actions).

File: .github/workflow/config.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      with:
+        node-version: '8.12.0'
+    - name: Cache Node.js modules
+      uses: actions/cache@v1
+      with:
+        path: ~/.npm
+        key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.OS }}-node-
+          ${{ runner.OS }}-
```
Comment (dhruvkb) [inline .github/workflow/config.yaml]: I'm not entirely sure what this does, but does running cache here preserve `node_modules/` for the tasks defined below?

File: .github/workflow/config.yaml (hunk 1 above)
Comment (dhruvkb) [inline .github/workflow/config.yaml]: Always leave a newline at the end of a file.

**Review by dhruvkb — CHANGES_REQUESTED** [review]
This is good work. A couple of improvements though:
- the directory should be `.github/workflows/`
- the file extension should be `.yml` (I know it's wrong but that's what is used across the project)
- the reviews following this comment

File: .github/workflow/config.yaml (hunk 3 above)
Comment (megha070) [reply .github/workflow/config.yaml]: Sure will go through this. Thanks a ton.
@dhruvkb and @hitesh1157

File: .github/workflow/config.yaml (hunk 1 above)
Comment (krhitesh) [reply .github/workflow/config.yaml]: Indeed it is. I thought of "testing" as in testing CI workflow based on assertion of every job's expected outcome and their actual result. That's what created a confusion.

But now it's clear to me.

File: .github/workflow/config.yaml (hunk 2 above)
Comment (hitesh1157) [reply .github/workflow/config.yaml]: Any specific version you have in mind?

File: .github/workflow/config.yaml (hunk 1 above)
Comment (krhitesh) [reply .github/workflow/config.yaml]: Done.

File: .github/workflow/config.yaml (hunk 4 above)
Comment (krhitesh) [reply .github/workflow/config.yaml]: It didn't at first but after 5a75d27, it does. See: 
Screenshot from 2020-03-28 19-39-39

File: .github/workflow/config.yaml (hunk 2 above)
Comment (krhitesh) [reply .github/workflow/config.yaml]: Done.

**Comment by krhitesh** [comment]
@dhruvkb Please take a look.

**Review by dhruvkb — APPROVED** [review]

**Comment by brylie** [comment]
@all-contributors please add @krhitesh for infra, test

**Pull request merged**